src/vs/platform/contextkey/browser/contextKeyService.ts

```diff
@@ -149,6 +159,8 @@ class ConfigAwareContextValuesContainer extends Context {
 			default:
 				if (Array.isArray(configValue)) {
 					value = JSON.stringify(configValue);
+				} else {
+					value = ConfigAwareContextValuesContainer._configContextObject;
```

Comment:
Is this "marker value" still needed? When getting an event for configured setting `foo.bar` should you rely on this having been run first?

Reply:
The marker value is to ensure truthy/falsy `when` conditions will work with object types. Alternatively, we could actually store the real value.